filament/backend/src/metal/MetalDriver.mm

```diff
@@ -521,15 +521,13 @@
 void MetalDriver::update2DImage(Handle<HwTexture> th, uint32_t level, uint32_t xoffset,
         uint32_t yoffset, uint32_t width, uint32_t height, PixelBufferDescriptor&& data) {
     auto tex = handle_cast<MetalTexture>(mHandleMap, th);
-    tex->load2DImage(level, xoffset, yoffset, width, height, data);
-    scheduleDestroy(std::move(data));
+    tex->load2DImage(level, xoffset, yoffset, width, height, std::move(data));
 }
 
 void MetalDriver::updateCubeImage(Handle<HwTexture> th, uint32_t level,
         PixelBufferDescriptor&& data, FaceOffsets faceOffsets) {
     auto tex = handle_cast<MetalTexture>(mHandleMap, th);
-    tex->loadCubeImage(data, faceOffsets, level);
-    scheduleDestroy(std::move(data));
+    tex->loadCubeImage(std::move(data), faceOffsets, level);
```

Comment:
The order of the parameters is different from load2DImage(), it'd be nicer if they were both consistent.